spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -130,34 +129,25 @@ public <T> T convertOnRead(
     return convertOnRead(val, embeddedType, collectionType, componentTypeInformation);
   }
 
+  @SuppressWarnings("unchecked")
   private <T> T convertOnRead(
       Object val,
       EmbeddedType embeddedType,
       Class targetCollectionType,
-      TypeInformation targetComponentType) {
+      TypeInformation<?> targetComponentType) {
     if (val == null) {
       return null;
     }
-    BiFunction<Object, TypeInformation<?>, ?> readConverter;
-    switch (embeddedType) {
-      case EMBEDDED_MAP:
-        readConverter =
-            (x, typeInformation) ->
-                convertOnReadSingleEmbeddedMap(
-                    x,
-                    typeInformation.getComponentType().getType(),
-                    typeInformation.getMapValueType(),
-                    targetComponentType);
-        break;
-      case EMBEDDED_ENTITY:
-        readConverter = this::convertOnReadSingleEmbedded;
-        break;
-      case NOT_EMBEDDED:
-        readConverter = this::convertOnReadSingle;
-        break;
-      default:
-        throw new DatastoreDataException("Unexpected property embedded type: " + embeddedType);
-    }
+    BiFunction<Object, TypeInformation<?>, ?> readConverter = switch (embeddedType) {
+      case EMBEDDED_MAP -> (x, typeInformation) ->
+          convertOnReadSingleEmbeddedMap(
+              x,
+              Objects.requireNonNull(typeInformation.getComponentType()).getType(),
+              typeInformation.getMapValueType(),
+              targetComponentType);
+      case EMBEDDED_ENTITY -> this::convertOnReadSingleEmbedded;
+      case NOT_EMBEDDED -> this::convertOnReadSingle;
+    };
```

Comment:
Great, adapting new language features! 
Is it safe to remove the default here? Seems that `embeddedType` can be passed in from the public `convertOnRead` method.

Reply:
I think the only possible value might have an issue here is `null` and `EmbeddedType.ordinal()` will throw a null pointer exception in that case.